support/spring/src/main/java/org/apache/shiro/spring/web/ShiroFilterFactoryBean.java

```diff
@@ -354,7 +354,7 @@ public void setGlobalFilters(List<String> globalFilters) {
      * @return the application's Shiro Filter instance used to filter incoming web requests.
      * @throws Exception if there is a problem creating the {@code Filter} instance.
      */
-    public Object getObject() throws Exception {
+    public AbstractShiroFilter getObject() throws Exception {
```

Comment:
I like this change, but this is API breaking.

@xczs666 any ideas on how to work around this as you were digging into the problem originally?

Reply:
@xczs666 Thinking about this again, for the 1.x line of Shiro. could we revert lines 357 and 120.  I'm guessing that line 370 is all that's needed?

That said, I wasn't able to reproduce the original issue to test it out either way.  Do you have a simple example project that demos the problem?  From there we should be able write a quick test around the problem.